src/algebra/fields/fp.tcc

```diff
@@ -197,11 +197,11 @@ Fp_model<n,modulus>::Fp_model(const long x, const bool is_unsigned)
 {
     if (is_unsigned || x >= 0)
     {
-        this->mont_repr.data[0] = x;
+        this->mont_repr.data[0] = (mp_limb_t)x;
```

Comment:
... or ``std::numeric_limits<mp_limb_t>::max() >= std::numeric_limits<long>::max()``, which correctly accounts for padding bits.